authors/apps/articles/views.py

```diff
@@ -92,3 +90,83 @@ def update(self, request, slug,*args,**kwargs):
         serializer.is_valid(raise_exception=True)
         serializer.save()
         return Response(serializer.data)
+
+
+class ShareFacebookView(ListCreateAPIView):
+    """creating links for sharing articles via facebook
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleShareLinkRenderer, )
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get("slug")
+        return get_object_or_404(Article, slug=slug)
+
+    def post(self, request, slug, *args, **kwargs):
+        self.get_object(slug)
```

Comment:
@frankopkusianwar, could you elaborate on  the use of this line her

Reply:
@kimbugp  It is to ensure the article of that slug exists as the user searches the url, but also since am required to add the title to the message ill use it to capture article attributes to add to the message